app/src/main/java/com/github/drawyourpath/bootcamp/userProfileCreation/UserGoalsInitFragment.kt

```diff
@@ -0,0 +1,137 @@
+package com.github.drawyourpath.bootcamp.userProfileCreation
+
+import android.graphics.Color
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import android.widget.EditText
+import android.widget.TextView
+import androidx.fragment.app.Fragment
+import com.github.drawyourpath.bootcamp.R
+import com.github.drawyourpath.bootcamp.database.Database
+import com.github.drawyourpath.bootcamp.database.FireDatabase
+import com.github.drawyourpath.bootcamp.database.MockDataBase
+
+class UserGoalsInitFragment : Fragment() {
+    private var isTest: Boolean = false
+    private var username: String = ""
+
+    //all this goals are per days
+    private var timeGoal: Int = 0
+    private var distanceGoal: Int = 0
+    private var nunberOfPathGoal: Int = 0
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        // Inflate the layout for this fragment
+        val view: View = inflater.inflate(R.layout.fragment_user_goals_init, container, false)
+
+        //retrieve the isRunTestValue and userName from the PersonalInfoFragment
+        val argsFromLastFrag: Bundle? = arguments
+        if (argsFromLastFrag == null) {
+            isTest = false
+        } else {
+            isTest = argsFromLastFrag.getBoolean("isRunningTestForDataBase")
+            username = argsFromLastFrag.getString("userName").toString()
+        }
+
+        //select the correct database in function of test scenario
+        var database: Database? = null
+        database = if (isTest) {
+            MockDataBase()
+        } else {
+            FireDatabase()
+        }
+        //all the goals inputs
+        val inputTimeGoal: EditText =
+            view.findViewById(R.id.input_timeGoal_text_UserProfileCreation)
+        val inputDistanceGoal: EditText =
+            view.findViewById(R.id.input_distanceGoal_text_UserProfileCreation)
+        val inputNbOfPathsGoal: EditText =
+            view.findViewById(R.id.input_nbOfPathsGoal_text_UserProfileCreation)
+
+        //all the texts where the potentials errors will be print
+        val errorTextTime: TextView = view.findViewById(R.id.timeGoalError_text_userProfileCreation)
+        val errorTextDistance: TextView =
+            view.findViewById(R.id.distanceGoalError_text_userProfileCreation)
+        val errorTextNbOfPaths: TextView =
+            view.findViewById(R.id.nbOfPathsGoalError_text_userProfileCreation)
+
+        //if all the data goals are correct that than set this data to the database associate to the username and show the next fragment,
+        //when click on the validate button
+        val validateButton: Button = view.findViewById(R.id.setUserGoals_button_userProfileCreation)
+        validateButton.setOnClickListener {
```

Comment:
Might be better to extract the button action to its own function to ease the code, or it might be ok if the tests are extracted.